client/state/notices/middleware.js

```diff
@@ -122,6 +122,12 @@ export const onPublicizeConnectionCreate = ( dispatch, { connection } ) => dispa
 	} ) )
 );
 
+export const onPublicizeConnectionCreateFailure = ( dispatch, { error } ) => dispatch(
+	errorNotice( error.message || translate( 'An error occurred while connecting the account.', {
```

Comment:
@obenland it looks like that error is coming from the server. Can you double check if it's safe to use here (eg i18n'). Even if it is safe, it may be worth while to switch case it to convert the error codes to translated messages in this middleware so we have a bit more control here.